### DISCUSSION TITLE
modal itration lead to open all models

### DISCUSSION THREAD

**Discussion by Adamkaram** [heroui-inc/heroui#1285] [Help]
currently i've used modal and itrate of it for 4 times and when i click in one of them they all opens at the same time even i specify a key for each of them 

![image](https://github.com/nextui-org/nextui/assets/52092726/5ce5bd5b-2b95-42f5-ae60-3725dcb1fa63)

code : https://stackblitz.com/edit/stackblitz-starters-wupqey?description=The%20React%20framework%20for%20production&file=pages%2Ftest.js&title=Next.js%20Starter

**Comment by tianenpang** [comment]
Hi @Adamkaram this is because the `Modal` used in the iteration is controlled by the same state. You could try separating the modal or place the modal outside the iterator and set the correct content state for the modal when it's opened.

Example of separating

```tsx
const MODALS = [
  { key: "m1", title: "Modal 1", content: <>...</> },
  { key: "m2", title: "Modal 2", content: <>...</> },
  { key: "m3", title: "Modal 3", content: <>...</> },
];


const MyModal = ({ data }) => {
  const {isOpen, onOpen, onOpenChange} = useDisclosure();

  return (
    <>
      <Button onPress={onOpen}>Open {data.title}</Button>
      <Modal isOpen={isOpen} onOpenChange={onOpenChange}>
        <ModalContent>
          {(onClose) => (
            <>
              <ModalHeader>{data.title}</ModalHeader>
              <ModalBody>
                {data.content}
              </ModalBody>
              <ModalFooter>
                <Button color="danger" variant="light" onPress={onClose}>
                  Close
                </Button>
              </ModalFooter>
            </>
          )}
        </ModalContent>
      </Modal>
    </>
  );
};

const App = () => {
  return (
    <>
      {MODALS.map((data) => <MyModal key={data.key} data={data} />)}
    </>
  );
};

```
